src/main/java/uk/gov/hmcts/reform/sscs/service/SubmitAppealServiceBase.java

```diff
@@ -145,125 +140,153 @@ public Long submitAppeal(SyaCaseWrapper appeal, String userToken) {
-                && rpc != null) {
+            && StringUtils.isNotEmpty(processingVenue)
+            && rpc != null) {
             String venueEpimsId = venueService.getEpimsIdForVenue(processingVenue);
             CourtVenue courtVenue = refDataService.getCourtVenueRefDataByEpimsId(venueEpimsId);
 
             sscsCaseData.setCaseManagementLocation(CaseManagementLocation.builder()
-                    .baseLocation(rpc.getEpimsId())
-                    .region(courtVenue.getRegionId()).build());
+                .baseLocation(rpc.getEpimsId())
+                .region(courtVenue.getRegionId()).build());
 
             log.info("Successfully updated case management location details for case {}. Processing venue {}, epimsId {}",
-                    appeal.getCcdCaseId(), processingVenue, venueEpimsId);
+                appeal.getCcdCaseId(), processingVenue, venueEpimsId);
         }
 
         log.info("{} - setting venue name to {}",
-                sscsCaseData.getAppeal().getAppellant().getIdentity().getNino(),
-                sscsCaseData.getProcessingVenue());
+            isIba
+                ? sscsCaseData.getAppeal().getAppellant().getIdentity().getIbcaReference()
+                : sscsCaseData.getAppeal().getAppellant().getIdentity().getNino(),
+            sscsCaseData.getProcessingVenue());
 
         return sscsCaseData;
     }
 
     private EventType findEventType(SscsCaseData caseData, boolean saveAndReturnCase) {
-
-        if (caseData.getAppeal().getMrnDetails() != null && caseData.getAppeal().getMrnDetails().getMrnDate() != null) {
-            LocalDate mrnDate = LocalDate.parse(caseData.getAppeal().getMrnDetails().getMrnDate());
+        Appeal appeal = caseData.getAppeal();
+        MrnDetails mrnDetails = appeal.getMrnDetails();
+        if (mrnDetails != null && mrnDetails.getMrnDate() != null) {
+            LocalDate mrnDate = LocalDate.parse(mrnDetails.getMrnDate());
             boolean moveToNoneCompliant = mrnDate.plusMonths(13L).isBefore(LocalDate.now());
 
             return handleMoveToNonCompliant(caseData, saveAndReturnCase, moveToNoneCompliant);
         } else {
-            log.info("Moving case for NINO {} to incomplete due to MRN Details {} present and MRN Date {} present",
-                    caseData.getAppeal().getAppellant().getIdentity().getNino(),
-                    (caseData.getAppeal().getMrnDetails() != null ? "" : "not"),
-                    (caseData.getAppeal().getMrnDetails().getMrnDate() != null ? "" : "not"));
+            boolean isIba = caseData.getBenefitCode().equals(Benefit.INFECTED_BLOOD_APPEAL.getBenefitCode());
+            Identity identity = appeal.getAppellant().getIdentity();
+            log.info("Moving case for {} {} to incomplete due to MRN Details {} present and MRN Date {} present",
+                isIba ? "Ibca Reference" : "NINO",
+                isIba ? identity.getIbcaReference() : identity.getNino(),
```

Comment:
do we want to log the Nino unmasked? it's PII info

Reply:
it was logged unmasked before, i can change this to log masked though